packages/cojson/src/coValueCore/coValueCore.ts

```diff
@@ -919,51 +944,20 @@ export class CoValueCore {
         return new Set();
       }
 
-      const dependentOn = this.getDependedOnCoValuesFromHeaderAndSessions(
+      const dependentOn = getDependedOnCoValuesFromRawData(
+        this.id,
         this.verified.header,
         this.verified.sessions.keys(),
+        Array.from(
+          this.verified.sessions.values(),
+          (session) => session.transactions,
+        ),
       );
       this._cachedDependentOn = dependentOn;
       return dependentOn;
     }
   }
 
-  /** @internal */
-  getDependedOnCoValuesFromHeaderAndSessions(
-    header: CoValueHeader,
-    sessions: Iterable<SessionID>,
-  ): Set<RawCoID> {
-    const deps = new Set<RawCoID>();
-
-    for (const session of sessions) {
-      const accountId = accountOrAgentIDfromSessionID(session);
-
-      if (isAccountID(accountId) && accountId !== this.id) {
-        deps.add(accountId);
-      }
-    }
-
-    if (header.ruleset.type === "group") {
-      if (isAccountID(header.ruleset.initialAdmin)) {
-        deps.add(header.ruleset.initialAdmin);
-      }
-
-      if (this.verified) {
-        for (const id of getGroupDependentKeyList(
-          expectGroup(this.getCurrentContent()).keys(),
```

Comment:
This was consuming the group content before, leading us to throw when the parent groups weren't available.

Refactored into a dependency check that reads only from the raw transactions.